README.md

```diff
@@ -0,0 +1,78 @@
+## Tango - (Trading Analysis & Go)
+
+[![Go Reference](https://pkg.go.dev/badge/github.com/jellydator/tango.svg)](https://pkg.go.dev/github.com/jellydator/tango)
+[![Build Status](https://github.com/jellydator/tango/actions/workflows/go.yml/badge.svg)](https://github.com/jellydator/tango/actions/workflows/go.yml)
+[![Go Report Card](https://goreportcard.com/badge/github.com/jellydator/tango)](https://goreportcard.com/report/github.com/jellydator/tango)
+
+## Features
+- Simple API
+- Built-in parameters validation
+- Includes thorough documentation
+- A wide variety of [Oscillators](#Oscillators) and [Overlays](#Overlays).
+
+## Installation
+```
+go get github.com/jellydator/tango
+```
+
+## Usage
+All of the tools must be created using `New*` function. It performs
+parameters validation and returns an object that is capable of working
+with data slices.
+
+The main calculations are done using `Calc` method. The return types varies
+based on the tool.
+
+A simple use case could look like this:
+```go
+func main() {
+  sma, err := tango.NewSMA(3)
+  if err != nil {
+    // handle the error.
+  }
+
+  dataPoints := []decimal.Decimal{
+    decimal.NewFromInt(2),
+    decimal.NewFromInt(3),
+    decimal.NewFromInt(4),
+  }
+
+  // the value is 3
+  value, err := sma.Calc(dataPoints)
+  if err != nil {
+    // handle the error.
+  }
+}
+```
+
+For the calculation to be successful, the `Calc` method should receive only the
+information that it requires. In some scenarios, it might not be known how many
+data points is needed, for this, a `Count` method may be used.
+
+```gosma
+func CalculateMA(ma tango.MA, values []decimal.Decimal) (decimal.Decimal, error) {
+  requiredValues := ma.Count()
+
+  if len(values) < requiredValues {
+    return decimal.Zero, errors.New("invalid count of values")
+  }
+
+  return ma.Calc(values[:requiredValues])
+}
+```
+
+## Oscillators
+- [Aroon](https://www.investopedia.com/terms/a/aroon.asp)
```

Comment:
I thought we could use links pointing to godoc, not sure which is better though.